OptimizelySDK.Tests/DecisionServiceTest.cs

```diff
@@ -441,5 +449,281 @@ public void TestGetVariationWithBucketingId()
+            
+            var userAttributes = new UserAttributes {
+                { "browser_type", "chrome" }
+            };
+
+            BucketerMock.Setup(bm => bm.Bucket(It.IsAny<ProjectConfig>(), It.IsAny<Experiment>(), It.IsAny<string>(), It.IsAny<string>())).Returns<Variation>(null);
+            var decisionService = new DecisionService(BucketerMock.Object, ErrorHandlerMock.Object, ProjectConfig, null, LoggerMock.Object);
+
+            var actualVariation = decisionService.GetVariationForFeatureRollout(featureFlag, "user_1", userAttributes);
+            Assert.IsNull(actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.DEBUG, $"Attempting to bucket user \"user_1\" into rollout rule \"{experiment.Key}\"."));
+            LoggerMock.Verify(l => l.Log(LogLevel.DEBUG, "User \"user_1\" is excluded due to traffic allocation. Checking \"Eveyrone Else\" rule now."));
+            LoggerMock.Verify(l => l.Log(LogLevel.DEBUG, $"User \"user_1\" is excluded from \"Everyone Else\" rule for feature flag \"{featureFlag.Key}\"."));
+        }
+        
+        // Should return expected variation when the user is attempted to be bucketed into all targeting rules
+        // including Everyone Else rule
+        [Test]
+        public void TestGetVariationForFeatureRolloutWhenUserDoesNotQualifyForAnyTargetingRule()
+        {
+            var featureFlag = ProjectConfig.GetFeatureFlagFromKey("boolean_single_variable_feature");
+            var rolloutId = featureFlag.RolloutId;
+            var rollout = ProjectConfig.GetRolloutFromId(rolloutId);
+            var experiment0 = rollout.Experiments[0];
+            var experiment1 = rollout.Experiments[1];
+            var everyoneElseRule = rollout.Experiments[rollout.Experiments.Count - 1];
+            var expectedVariation = everyoneElseRule.Variations[0];
+
+            BucketerMock.Setup(bm => bm.Bucket(It.IsAny<ProjectConfig>(), everyoneElseRule, It.IsAny<string>(), It.IsAny<string>())).Returns(expectedVariation);
+            var decisionService = new DecisionService(BucketerMock.Object, ErrorHandlerMock.Object, ProjectConfig, null, LoggerMock.Object);
+
+            // Provide null attributes so that user does not qualify for audience.
+            var actualVariation = decisionService.GetVariationForFeatureRollout(featureFlag, "user_1", null);
+            Assert.AreEqual(expectedVariation, actualVariation);
+
+            LoggerMock.Verify(l => l.Log(LogLevel.DEBUG, $"User \"user_1\" does not meet the audience conditions to be in rollout rule \"{experiment0.Key}\"."));
+            LoggerMock.Verify(l => l.Log(LogLevel.DEBUG, $"User \"user_1\" does not meet the audience conditions to be in rollout rule \"{experiment1.Key}\"."));
+        }
+
+        #endregion // GetVariationForFeatureRollout Tests
+
+        #region GetVariationForFeature Tests
+
+        // Should return the bucketed experiment and variation 
```

Comment:
minor nit: make test case comments more readable by stating conditions
example: Should return the variation the user was bucketed into when the user is in the feature flag's experiment.

Reply:
Done.